libkbfs/data_types.go

```diff
@@ -941,3 +941,8 @@ func SessionInfoFromProtocol(session keybase1.Session) (SessionInfo, error) {
 		VerifyingKey:   verifyingKey,
 	}, nil
 }
+
+// NodeMetadata has metadata about a node needed for higher level operations.
+type NodeMetadata struct {
```

Comment:
Actually I guess `DirEntry.EntryInfo` isn't useful, since all that is available with `ls -l` already.  But anything in `DirEntry.BlockInfo` is potentially interesting.